templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -25,7 +25,8 @@ contents:
             logger -s "NM resolv-prepender: Prepending 'nameserver $NAMESERVER_IP' to /etc/resolv.conf (other nameservers from /var/run/NetworkManager/resolv.conf)"
             sed -e "/^search/d" \
                 -e "/Generated by/c# Generated by KNI resolv prepender NM dispatcher script\nsearch $DOMAIN\nnameserver $NAMESERVER_IP" \
-                /var/run/NetworkManager/resolv.conf > /etc/resolv.conf
+                /var/run/NetworkManager/resolv.conf > /etc/resolv.tmp
+            mv -f /etc/resolv.tmp /etc/resolv.conf
```

Comment:
There's a sed below at well that limits us to the top 3 nameservers. Should we do it all (including that sed) in a temp file and then copy the file result?

Reply:
You're right, we should. That fits well with cgwalters suggestion below as well.